### PR TITLE
Feature/issue 56

### PR DESCRIPTION
**Pull request by raynasr**
Refactored the entire UI structure. 
Check presentation and presentation.ui
Check usecase.user

### CODE REVIEW

**Review by edouardmichelin — CHANGES_REQUESTED** [review]
Some files need to be removed.
Some changes are requested regarding conventions and project architecture.
Otherwise good job!

**Comment by edouardmichelin** [comment]
Such a big PR goes against all my beliefs but its purpose is to refactor a big part of the project and it had to be done

File: app/src/main/java/com/studhub/app/presentation/home/HomeScreen.kt [hunk 1]
```diff
@@ -21,63 +20,48 @@ fun HomeScreen(
...
-                text =
-                if (user.value == null)
-                    "Welcome to our app!"
-                else
-                    "Welcome to our app ${user.value!!.userName}!"
-            )
-
-            Spacer(Modifier.height(16.dp))
-            Text(text = "Here are some featured items:")
+            WelcomeText(viewModel = viewModel)
```
[the remark this answers is not in the document]
Comment (Edouard-6) [reply app/src/main/java/com/studhub/app/presentation/home/HomeScreen.kt:34]: I agree with you that we can do some exceptions, for example like here and pass the viewModel directly.

File: app/src/main/res/values/strings_about.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="about_title">About us:</string>
+    <string name="about_content">We are a MarketPlace company working for EPFL campus.</string>
+    <string name="about_contact_title">Contact us:</string>
+    <string name="about_contact_content">contact@studhub.ch</string>
+</resources>
```
Comment (Edouard-6) [inline app/src/main/res/values/strings_about.xml:7]: Add newline at end of file.

File: app/src/main/res/values/strings_home.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<resources>
+    <string name="home_title">Home page</string>
+    <string name="home_welcome_message">Welcome to our app</string>
+    <string name="home_welcome_name_message">Welcome to our app %s.</string>
+    <string name="home_featured_items_title">Here are some feature items</string>
+    <string name="home_button_add_listing">Add Listing</string>
+    <string name="home_button_browse">Browse</string>
+    <string name="home_button_cart">Cart</string>
+    <string name="home_button_about">About</string>
+</resources>
```
Comment (Edouard-6) [inline app/src/main/res/values/strings_home.xml:11]: Add newline at end of file.

File: app/src/main/res/values/strings_listings.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="listings_add_title">Create your Listing</string>
+    <string name="listings_browsing_title">Browse all listings</string>
+</resources>
```
Comment (Edouard-6) [inline app/src/main/res/values/strings_listings.xml:5]: Add newline at end of file.

File: app/src/main/res/values/strings_misc.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="misc_currency_symbol">CHF</string>
+</resources>
```
Comment (Edouard-6) [inline app/src/main/res/values/strings_misc.xml:4]: Add newline at end of file.

File: gradle.properties [hunk 6]
```diff
@@ -23,4 +23,5 @@ android.nonTransitiveRClass=true
 org.gradle.daemon=true
 org.gradle.caching=true
 org.gradle.parallel=true
-org.gradle.configureondemand=true
```
Comment (Edouard-6) [inline gradle.properties:26]: Add newline at end of file.

**Pull request merged**